### PR TITLE
Fixes right click with multiple selected assets in the thumbnail view

### LINKED ISSUE #15921: Asset Browser: Right-clicking in the Thumbnail View with multiple assets selected only operates on the last selected asset
**Describe the bug**
Right-clicking in the Thumbnail View with multiple assets selected only operates on the last selected asset.

**Steps to reproduce**
1. Launch the Editor.
2. Open a new or existing level.
3. In the Asset Browser, switch to Thumbnail View.
4. Navigate to a directory with multiple assets.
5. CTRL-Click to select multiple assets.
6. Right-click one of the selected assets.

**Expected behavior**
The context menu action represents all of the selected assets.

**Actual behavior**
Only the right-clicked asset is operated on. The previously selected assets are deselected. This only occurs in Thumbnail View. If the user continues to hold CTRL when right-clicking, the context menu appropriately operates on all assets selected, but this only occurs in Thumbnail View.

**Found in Branch**
development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
d7ea7f7ee597079d7e246fd50f76a550a1f774d7

### PR DESCRIPTION
**Pull request by dtamkin1**
## What does this PR do?
- Fixes #15921 
- Fixes keyboard shortcuts not registering in the table view
- Fixes assets not prompting user for rename when newly created in the table view

## How was this PR tested?

Locally

### CODE REVIEW

File: Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp [hunk 1]
```diff
@@ -790,9 +790,10 @@ namespace AzQtComponents
         auto idx = indexAtPos(p);
         if (idx.isValid())
         {
+            bool isRightClick = event->button() == Qt::RightButton;
             if (selectionMode() == ExtendedSelection && QGuiApplication::queryKeyboardModifiers() == Qt::ControlModifier)
             {
-                selectionModel()->select(idx, QItemSelectionModel::SelectionFlag::Toggle);
+                (isRightClick ? selectionModel()->select(idx, QItemSelectionModel::SelectionFlag::Select) : selectionModel()->select(idx, QItemSelectionModel::SelectionFlag::Toggle));
```
Comment (forkercat) [inline Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp:796]: nit: Would be good to do this instead to improve readability a bit.

```cpp
auto selectionFlag = isRightClick ? QItemSelectionModel::SelectionFlag::Select : QItemSelectionModel::SelectionFlag::Toggle;
selectionModel()->select(idx, selectionFlag);
```

**Review by forkercat — APPROVED** [review]

File: Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp (hunk 1 above)
Comment (gadams3) [reply Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp:796]: I agree. While this should work, you should either explicitly use an if statement and curly braces for the function calls or move the condition into the second parameter for the selection flag. The above recommendation is the cleanest.

File: Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp [hunk 2]
```diff
@@ -814,11 +815,12 @@ namespace AzQtComponents
                                                            : QItemSelectionModel::SelectionFlag::Select);
                     for (auto index : indexRange.indexes())
                     {
-                        selectionModel()->select(index, selectionFlag);
+                        (isRightClick ? selectionModel()->select(index, QItemSelectionModel::SelectionFlag::Select)
+                                      : selectionModel()->select(index, selectionFlag));
```
Comment (gadams3) [inline Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp:819]: Same comment as above about expanding or reformatting this condition

**Review by gadams3 — CHANGES_REQUESTED** [review]

File: Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp (hunk 1 above)
Comment (dtamkin1) [reply Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp]: Done :)

File: Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp (hunk 2 above)
Comment (dtamkin1) [reply Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp]: Done!

**Review by cgalvan — APPROVED** [review]

**Review by gadams3 — APPROVED** [review]

**Pull request merged**